### PR TITLE
Pimcore X Upgrade Guide

### LINKED ISSUE #8109: Pimcore X Upgrade Guide from v6.9
- Flex folder structure
- `security.yaml` - config is not merged anymore (see #8186 )

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #8109

### CODE REVIEW

File: doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md [hunk 1]
```diff
@@ -1,12 +1,57 @@
-# Upgrading Pimcore from Version 6.9 to Version 10
+# Upgrading Pimcore from Version 6.x to Version 10
 
-# Flush your `/tmp` directories
+## System Requirements
+ - PHP >= 8.0
+ - Apache >= 2.4
```
Comment (jdreesen) [inline doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md:5]: Is `Apache` really required? We use `nginx`, for example ;)

File: doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md [hunk 2]
```diff
@@ -1,12 +1,57 @@
...
+
+### Cleanup your composer.json
+Update composer.json as per [skeleton](https://github.com/pimcore/skeleton/blob/master/composer.json)
+
+### Update folder structure 
+Update your project folder structure as per [Symfony Flex](https://symfony.com/doc/current/setup/flex.html) guidelines. See demo changes [here](https://github.com/pimcore/demo/pull/203/files)
+
+Important changes:
+- Move configs from app/config/*.yml to config/*.yaml
+- Move templates from app/Resources/views to templates and bundles templates from
```
Comment (jdreesen) [inline doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md:28]: Something is missing here...

Comment (NiklasBr) [reply doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md:28]: I think it is clearer if it is rewritten and formatted:

> - Move templates from bundles and `app/Resources/views` to `templates`

File: doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md (hunk 1 above)
Comment (dvesh3) [reply doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md:5]: well, this is about what has changed on Pimcore X 😉 changed headings e.g. `System Requirement changes`

File: doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md (hunk 2 above)
Comment (dvesh3) [reply doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/10_V6_to_V10.md]: done!

**Pull request merged**